lib/open_feature/hook_context.rb

```diff
@@ -0,0 +1,27 @@
+# typed: strict
+# frozen_string_literal: true
+
+module OpenFeature
+  # See https://openfeature.dev/specification/sections/hooks#41-hook-context
+  # See Requirement 4.1.1, 4.1.3, 4.1.4
+  # TODO: Requirement 4.1.2
+  class HookContext < T::Struct
+    extend T::Generic
+    extend T::Sig
+
+    include T::Struct::ActsAsComparable
+
+    Value = type_member
+    const :flag_key, String
+    const :flag_type, String
+    const :evaluation_context, EvaluationContext
+    const :default_value, Value
```

Comment:
I think we should introduce [Requirement 4.1.2](https://openfeature.dev/specification/sections/hooks#requirement-412)

```suggestion
    const :default_value, Value
    const :client_metadata, ClientMetadata
    const :provider_metadata, ProviderMetadata
```